### ISSUE TITLE
Handle webview touches consistently, with TouchEvent

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-mobile#3439]
Our handling of touches and clicks inside the message-list WebView is rather ad hoc. It's harder to understand than it should be (though #3434 improves the situation). It's especially hard to reason about what happens if there are several touches, and it seems likely there is some surprising behavior in there.

Happily, the Web API that we're using makes it possible to handle these in a fairly clean way! We just need to do so.

A key part of this task will be to read and understand the documentation about `TouchEvent`; the various `touch...` event types that we can listen to; and the relationship between touch events and `click` events.
* MDN on touch events in general: [guide](https://developer.mozilla.org/en-US/docs/Web/API/Touch_events/Using_Touch_Events), [example](https://developer.mozilla.org/en-US/docs/Web/API/Touch_events/Multi-touch_interaction), and API reference for [TouchEvent](https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent) and [Touch](https://developer.mozilla.org/en-US/docs/Web/API/Touch)
* MDN on [TouchEvent and MouseEvent](https://developer.mozilla.org/en-US/docs/Web/API/Touch_events/Supporting_both_TouchEvent_and_MouseEvent); for this bit, I found I had to also read the relevant [spec section](https://w3c.github.io/touch-events/#mouse-events) before I felt I understood

Some specific observations, from the docs on touch events in general:
* We don't listen to `touchcancel`. This probably is a bug.
* We remember just one touch position, `lastTouchPosition{X,Y}`, even when several touches happen. On subsequent events, we effectively assume that they're about the same touch... except there's that `isNearPositions` condition in the `touchend` handler, one effect of which is to try to check if the touch that's ending is the same one as we remember. (A separate effect is that long-presses that move around don't count -- I think this is the main intended effect.)
* We look at only one element of the `changedTouches` array, rather than iterating through them. In the `touchend` handler, this means that if both a touch we're ignoring and the one we currently remember end in the same event... depending on the order, we could completely miss the fact that the one we were paying attention to ended.

In general I think this can all be cleaned up mainly by just using the `identifier` property of the `Touch` objects so we know the difference between a new touch and the same touch.

Some specific observations, from the interaction with `click`:

* We could invoke the logic that's in our `click` handler straight from the `touchend` handler, and suppress the automatic `click`. That would make the logic for normal and long presses much more similar, which would I think help quite a bit in making it clearer in the code (for our own understanding now, and for people reading in the future) how it works.

* It's currently impossible for a user to get to our long-press-based features if they have a pointing device that doesn't do touch events. Is this a real situation? The spec is pretty stern about saying that it is:

  > Particularly in the case of touch-enabled laptops, or traditional "touch only" devices (such as phones and tablets) with paired external input devices, users may use the touchscreen in conjunction with a trackpad, mouse or keyboard.

  So e.g. someone could be using the app on a tablet, and sometimes have a keyboard and mouse, and it'd be frustrating if for some features they had to reach up to the screen.

  Fortunately, it looks like it should also be simple to support both. For `click`, we can just keep listening to `click` -- the `touchend` and `click` handlers can both just invoke a common function that has basically the body of our current `click` handler.

  And for long-presses... perhaps [`auxclick`](https://developer.mozilla.org/en-US/docs/Web/API/Element/auxclick_event) would be most natural?

**Comment by gnprice** [comment]
Update: with 1e8023741, we now have a `touchcancel` listener!

**Comment by chrisbobbe** [comment]
Potentially related: https://github.com/zulip/zulip-mobile/issues/3557.
